script.js

```diff
@@ -86,48 +86,55 @@ var gameBasic = function (input) {
   if (gameState == "playerRoll") {
     playersDice[currentPlayer] = randomDice();
     message = `Welcome Player ${currentPlayer}.<br>`;
-    message += `You rolled ${playersDice[currentPlayer]}.<br>`;
+    message += `You rolled ${playersDice[currentPlayer]}.<br><br>`;
     gameState = "playerVal";
     return message;
   }
 
   if (gameState == "playerVal") {
     playersScore[currentPlayer] = calculateScore(playersDice[currentPlayer]);
+    message = `Player ${currentPlayer}:<br>`;
     message += `Your number is ${playersScore[currentPlayer]}.<br>`;
     if (currentPlayer < playersNum) {
       message += `It is now Player ${currentPlayer + 1}'s turn.`;
       currentPlayer += 1;
       gameState = "playerRoll";
+      return message;
     }
     gameState = "score";
     return message;
   }
 
   if (gameState == "score") {
     message = "";
-    if (player1Score == player2Score) {
+    var bestPlayerIndex = 1;
+    var bestPlayerScore = playersScore[1];
+    var tieGame = false;
+    for (var i = 1; i < playersNum + 1; i += 1) {
+      if (playersScore[i] == bestPlayerScore && i > 1) {
+        tieGame = true;
+      }
+      if (
+        (playersScore[i] > bestPlayerScore && gameType == "basic") ||
+        (playersScore[i] < bestPlayerScore && gameType == "lowest")
+      ) {
+        bestPlayerScore = playersScore[i];
+        bestPlayerIndex = i;
+      }
+    }
+
+    if (tieGame) {
       message += `It's a tie.<br><br>`;
-    } else if (
-      (player1Score > player2Score && gameType == "basic") ||
-      (player1Score < player2Score && gameType == "lowest")
-    ) {
-      player1Won += 1;
-      message += `Player 1 won this round with ${player1Score} vs ${player2Score}!<br><br>`;
     } else {
-      player2Won += 1;
-      message += `Player 2 won this round with ${player2Score} vs ${player1Score}!<br><br>`;
+      playersWins[bestPlayerIndex] += 1;
+      message += `Player ${bestPlayerIndex} won this round with ${playersScore[bestPlayerIndex]}!<br><br>`;
     }
     totalGames += 1;
-    var leaderBoard = [
```

Comment:
No leaderboard in variable player mode? Maybe want to think about how to implement that for the next project!